unified-codes/apps/web/src/components/containers/error/ErrorPage.tsx

```diff
@@ -1,34 +1,63 @@
+
 import * as React from 'react';
+import { useParams } from 'react-router-dom';
+
+import { makeStyles, createStyles } from '@unified-codes/ui/styles';
+
+import ErrorNotFoundTop from './ErrorNotFoundTop';
+import ErrorNotFoundCenter from './ErrorNotFoundCenter';
+import ErrorNotFoundBottom from './ErrorNotFoundBottom';
+
+import ErrorLayout from '../../layout/ErrorLayout';
+
+import { Errors, IErrorRouteParams } from '../../../types';
+import { ITheme } from '../../../styles';
+
+const ERROR_CODES = {
+  [Errors.PageNotFound]: '404',
+}
 
-import { Grid } from '@unified-codes/ui/components';
+const useStyles = makeStyles((_: ITheme) => createStyles({
+  root: {
+      flexDirection: 'column',
+      alignContent: 'center',
+      padding: '20px 0 20px 0',
+  },
+  centerContainer: {
+      flexDirection: 'row',
+      justifyContent: 'center',
+  },
+  topContainer: {
+      textAlign: 'center'
+  },
+  bottomContainer: {
+    textAligh: 'center'
+  }
+}));
 
-const CatOnBike = require('../../../assets/cat-on-bike.png');
 
 export interface ErrorPageProps {
-  code: string;
+  onMount?: (params: IErrorRouteParams) => void;
+  onUnmount?: (params: IErrorRouteParams) => void;
 }
 
 export type ErrorPage = React.FunctionComponent<ErrorPageProps>;
 
-export const ErrorPage: ErrorPage = ({ code }) => {
-  switch (code) {
-    case '404':
-      return (
-        <Grid container style={{ flexDirection: 'column', alignContent: 'center' }}>
-          <Grid item style={{ textAlign: 'center' }}>
-            <h3>Nothing to see here...</h3>
-          </Grid>
-          <Grid item>
-            <img style={{ maxHeight: 'calc(100vh - 300px)', maxWidth: '100vw ' }} src={CatOnBike} />
-          </Grid>
-          <Grid item style={{ textAlign: 'center' }}>
-            <h3>...are you sure that you're in the right place?</h3>
-          </Grid>
-        </Grid>
-      );
-    default:
-      return <div>Oops! Something went wrong.</div>;
+export const ErrorPage: ErrorPage = ({ onMount = () => null, onUnmount = () => null }) => {
+  const classes = useStyles();
+  const params: IErrorRouteParams = useParams();
+
+  React.useEffect(() => {
+      onMount(params);
+      return () => onUnmount(params);
+  }, []);
+
+  const { code = ERROR_CODES[Errors.PageNotFound] } = params;
+
+  switch(code) {
+    case ERROR_CODES[Errors.PageNotFound]: return <ErrorLayout classes={classes} top={<ErrorNotFoundTop/>} center={<ErrorNotFoundCenter/>} bottom={<ErrorNotFoundBottom/>} />
```

Comment:
I think the `ErrorNotFoundTop` `ErrorNotFoundCenter` and `ErrorNotFoundBottom` are OTT - why not just have an `ErrorNotFoundPage`? Or even an error page where the image and messaging is specific to the error and sourced somehow?

Reply:
I agree 100%... that was what I did first! I was worried about confusing people since we've implemented a bit of a page convention. I was hoping that as we add more error pages we will find that they share a common layout structure (e.g. maybe each error will have a shared image but different text or something). 

The reason I avoided `ErrorXPage` was to avoid confusion with the "page" convention (i.e. a page is something which is routed to from the main app and which has `onMount` and `onUnmount` props which take the navigation props). But I think as I look at it again, it's probably best to just has specific error pages.

Reply:
I think maybe for now we put up with this agreed OTT solution, and maybe when more error components are added, we can either adjust the layout to be a bit less redundant, or revert to having a `ErrorXPage` for each error page?

Reply:
yup, sounds good to me